widgets/tools/commentstab.h

```diff
@@ -40,12 +40,12 @@ Q_OBJECT
 
 public:
     void fill();
-    virtual int rowCount(const QModelIndex & parent = QModelIndex()) const override;
-    virtual int columnCount(const QModelIndex & parent = QModelIndex()) const override;
-    virtual QVariant headerData(int section, Qt::Orientation orientation, int role = Qt::DisplayRole) const override;
-    virtual QVariant data(const QModelIndex & index, int role = Qt::DisplayRole) const override;
-    virtual bool setData(const QModelIndex & index, const QVariant & value, int role = Qt::EditRole) override;
-    virtual Qt::ItemFlags flags(const QModelIndex & index) const override;
+    int rowCount(const QModelIndex & parent = QModelIndex()) const override;
+    int columnCount(const QModelIndex & parent = QModelIndex()) const override;
+    QVariant headerData(int section, Qt::Orientation orientation, int role = Qt::DisplayRole) const override;
+    QVariant data(const QModelIndex & index, int role = Qt::DisplayRole) const override;
+    bool setData(const QModelIndex & index, const QVariant & value, int role = Qt::EditRole) override;
+    Qt::ItemFlags flags(const QModelIndex & index) const override;
```

Comment:
i don’t know why it wants to remove all the `virtual` here, i guess `override` can already tell you it’s virtual
i personally like all virtual functions to be marked as such explicitly, and `override` if it’s 2nd+ level

Reply:
It suggests to use virtual only in the base class, and in all derived classes override. Using both produces the warning, that this is redundant.

Reply:
i don’t like this, but it comes from the fact that there were different driving factors behind `override` (specifically it’s position in the declaration) which bite here